src/classlibnative/bcltype/number.cpp

```diff
@@ -2109,13 +2109,8 @@ FCIMPL3_VII(Object*, COMNumber::FormatDouble, double value, StringObject* format
     int precision = DOUBLE_PRECISION;
     switch (val) {
     case 'R':
-        //In order to give numbers that are both friendly to display and round-trippable,
-        //we parse the number using 15 digits and then determine if it round trips to the same
-        //value.  If it does, we convert that NUMBER to a string, otherwise we reparse using 17 digits
-        //and display that.
-
-        DoubleToNumber(value, DOUBLE_PRECISION, &number);
-
+        // Per IEEE double-precision floating-point format, double number in 17 digits precision should be round-trippable.
```

Comment:
It might be useful to quote the exact IEEE statement here: 

> Conversions from a supported binary format bf to an external character sequence and back again results in a copy of the original number so long as there are at least Pmin(bf) significant digits specified and the rounding-direction attributes in effect during the two conversions are round to nearest rounding-direction attributes

With an additional note that, for binary64 (double), `Pmin(binary64)` is 17. The algorithm given is `Pmin(bf) = 1 + ceiling(p * log10(2))` where `p` is the number of significant bits in `bf` (this is 53 for binary64).